### DISCUSSION TITLE
How can I have management UI and DoH serve from different port?

### DISCUSSION THREAD

**Discussion by yegle** [AdguardTeam/AdGuardHome#7511] [Q&A]
As far as I can tell, there's only one `httpsServer` in the code:

https://github.com/AdguardTeam/AdGuardHome/blob/dedbadafc4a4d3c75e15206ba4fb05401824e950/internal/home/web.go#L308

What if I want to listen the management UI and the DoH on two different IP addresses? How can I make this to work?

For context: I'd like to expose https port for DoH to the internet, while accessing the management UI via a Tailscalle IP. AFAIKT, once you expose the 443 port for DoH to the internet, the same port can be used to access the management UI.

**Comment by AjieDev** [comment]
Im also looking for this, incase an "Brute-Force" Attack for my Public DNS Server

Edit, anyway we can use reverse proxy

**Reply by yegle** [reply]
Yep this is currently my solution, with a Caddyfile that looks like this:

```$ cat Caddyfile
(tls) {
        tls cert.pem key.pem
}

$ADGUARD_HOST {
        import tls
        @tailscale client_ip 100.64.0.0/10 172.16.0.0/12
        reverse_proxy @tailscale adguard:80
}
$DOH_HOST {
        import tls
        handle /dns-query* {
                reverse_proxy adguard:80
        }
}
```
